FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -0,0 +1,657 @@
+    for (key, value) in expectedCustomMetadata {
+      XCTAssertEqual(actualMetadata.customMetadata![key], value)
+    }
+  }
+
+  private func assertMetadataNil(actualMetadata: StorageMetadata) {
+    XCTAssertNil(actualMetadata.cacheControl)
+    XCTAssertNil(actualMetadata.contentDisposition)
+    XCTAssertEqual(actualMetadata.contentEncoding, "identity")
+    XCTAssertNil(actualMetadata.contentLanguage)
+    XCTAssertNil(actualMetadata.contentType)
+    XCTAssertTrue(actualMetadata.md5Hash!.count == 24)
+    XCTAssertNil(actualMetadata.customMetadata)
+  }
+
+  func testUpdateMetadata() {
+    let expectation = self.expectation(description: "testUpdateMetadata")
+    let ref = storage?.reference(withPath: "ios/public/1mb")
+
+    let metadata = StorageMetadata()
+    metadata.cacheControl = "cache-control"
+    metadata.contentDisposition = "content-disposition"
+    metadata.contentEncoding = "gzip"
+    metadata.contentLanguage = "de"
+    metadata.contentType = "content-type-a"
+    metadata.customMetadata = ["a": "b"]
+
+    ref?.updateMetadata(metadata, completion: { updatedMetadata, error in
+      XCTAssertNil(error, "Error should be nil")
+      guard let updatedMetadata = updatedMetadata else {
+        XCTFail("Metadata is nil")
+        return
+      }
+      self.assertMetadata(actualMetadata: updatedMetadata,
+                          expectedContentType: "content-type-a",
+                          expectedCustomMetadata: ["a": "b"])
+
+      let metadata = updatedMetadata
+      metadata.contentType = "content-type-b"
+      metadata.customMetadata = ["a": "b", "c": "d"]
+
+      ref?.updateMetadata(metadata, completion: { updatedMetadata, error in
+        XCTAssertNil(error, "Error should be nil")
+        self.assertMetadata(actualMetadata: updatedMetadata!,
+                            expectedContentType: "content-type-b",
+                            expectedCustomMetadata: ["a": "b", "c": "d"])
+        guard let metadata = updatedMetadata else {
+          XCTFail("Metadata is nil")
+          return
+        }
+        metadata.cacheControl = nil
+        metadata.contentDisposition = nil
+        metadata.contentEncoding = nil
+        metadata.contentLanguage = nil
+        metadata.contentType = nil
+        metadata.customMetadata = Dictionary()
+        ref?.updateMetadata(metadata, completion: { updatedMetadata, error in
+          XCTAssertNil(error, "Error should be nil")
+          self.assertMetadataNil(actualMetadata: updatedMetadata!)
+          expectation.fulfill()
+        })
+      })
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedResumeGetFile() {
+    let expectation = self.expectation(description: "testUnauthenticatedResumeGetFile")
+    let ref = storage?.reference(withPath: "ios/public/1mb")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+
+    let task = ref?.write(toFile: fileURL)
```

Comment:
Nit: `guard let task = ...` to avoid unwrapping it each time and stopping the test earlier in the case of an error.

Reply:
Done.